addons/survey/models/survey_survey.py

```diff
@@ -355,6 +359,40 @@ def _compute_scoring_type(self):
             elif not survey.scoring_type:
                 survey.scoring_type = 'no_scoring'
 
+    @api.depends('survey_type')
+    @api.depends_context('uid')
+    def _compute_allowed_survey_types(self):
+        # Survey user can't see/select survey types added from other apps
+        # Whereas survey manager can see/use them all
+        self.allowed_survey_types = self._fields["survey_type"].selection \
+            if self.env.user._has_group('survey.group_survey_manager') \
+            else [choice for choice in self._fields["survey_type"].selection
+                  if choice[0] in {'assessment', 'custom', 'live_session', 'survey'}]
+
+    @api.depends_context('uid')
+    def _compute_is_survey_restriction_allowed(self):
+        # compute based on restricted users
+        restricted = self.filtered('restrict_user_ids') if not self.env.user.has_group('survey.group_survey_manager') else self.browse()
```

Comment:
While I'm here I'm showing you this pattern to be more straightforward and readable:
("while I'm here" because I'm not sure to follow how this method works...)

```py
if self.env.user.has_group('survey.group_survey_manager'):
    self.is_survey_restriction_allowed = False
    return
```